keps/sig-auth/4193-tokenrequest-requester-info-enhancements/README.md

```diff
@@ -0,0 +1,596 @@
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+* `ServiceAccountTokenJTI` feature flag will toggle including JTI information in tokens, as well as recording JTIs in the audit log.
+* `ServiceAccountTokenNodeInfo` feature flag will toggle including node info in tokens.
+
+Both of these feature flags can be disabled without any unexpected adverse affects or coordination required.
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `ServiceAccountTokenJTI`
+  - Components depending on the feature gate: kube-apiserver
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `EmbedServiceAccountRequesterInfo`
+  - Components depending on the feature gate: kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+Enabling the feature gate will cause additional information to be stored/persisted into service account JWTs, as well
+as new audit annotations being recorded in the audit log. This is all purely additive, so no changes to existing
+features, schemas or fields are expected.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Future tokens will then not embed this information. Any existing issued tokens **will** still have this
+information embedded however.
+
+If these fields are deemed to be problematic for other systems interpreting these tokens, users will need to re-issue
+these tokens before presenting them elsewhere.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Future tokens will once again include this information/no adverse effects.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes (as noted above in the test plan)
+
+### Rollout, Upgrade and Rollback Planning
+
+Rolling this out will be done by enabling the feature flag on all control plane hosts.
+This should not have any issues/affect during upgrades.
+Rollback is done by removing/disabling the feature gate.
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+If somehow the tokens issued during the time where the feature was enabled were to be rejected, these tokens would
+need to be re-requested.
+
+This isn't so much a failure of the rollback, rather it would be a bug in the actual feature (hence feature gating).
+
+During a skew where hosts have different enablement, users would get tokens with varying content if they made multiple
+requests. These tokens will all be usable against all apiservers though, so there won't be issues due to skew.
```

Comment:
We should probably manually test and confirm that a new token with node ref and JTI has no negative impact on a v1.28 API server (and that it still works).